File: Strings/Palindromes/6a/isPalin.js
var isPalindrome = function (s) {
  if (s.length <= 1) return true;

  const regex = /^[a-z0-9A-Z]+$/;
  let parsedString = [];
  for (let i=0; i < s.length; i++) {
    if (s[i].match(regex)) {
      parsedString.push(s[i].toLowerCase())
    }
  }
  
  let i = 0, j = parsedString.length -1;
  while (i < j) {
    if (parsedString[i] !== parsedString[j]) return false;

    i++; j--;
  }
  return true;
};

s = 'madam';
console.log(isPalindrome(s));

File: Strings/Palindromes/6a/isPalin.py
def isPalindrome(s):
    if len(s) <= 1:
        return True
    parsedString = [string.lower() for string in s if string.isalnum()]
    p1, p2 = 0, len(parsedString) - 1
    while p1 < p2:
        if parsedString[p1] != parsedString[p2]:
            return False
        p1 += 1
        p2 -= 1
    return True

print(isPalindrome("a race car"))
